app/mailers/api_entreprise_mailer.rb

```diff
@@ -1,3 +1,4 @@
 class APIEntrepriseMailer < ApplicationMailer
   default from: 'support@entreprise.api.gouv.fr'
+  default_url_options[:host] = 'https://entreprise.api.gouv.fr'
```

Comment:
Je ne sais pas si c'est override en sandbox/staging/prod, à voir/tester (imo non va falloir le faire de manière plus intelligente, genre faire un helper)

Reply:
WDYT? https://github.com/etalab/admin_api_entreprise/pull/1255/commits/1330ffc80833975b582b3312f3db502d3bd80a9f

Reply:
Mh c'est con mais j'ai du mal à faire un helper ici. Je fais ça mais je sais pas trop où le caser:

```
  def host_url_for(api:)
    return "https://#{Rails.env}.#{api}.api.gouv.fr" if Rails.env.sandbox? || Rails.env.staging?

    "https://#{api}.api.gouv.fr"
  end
```

et  `default_url_options[:host] = host_url_for(api: 'entreprise')`

Reply:
Je pense que tu configures ça niveau rails app entière. Exemple: https://www.benpickles.com/articles/70-dynamically-setting-rails-default_url_options-in-heroku-review-apps

Reply:
Bon j'ai fait un truc dans le dernier commit, c'est pas ultra-top mais j'ai flemmé de ouf, tu me dis si tu as une meilleure suggestion

Reply:
Test ça https://stackoverflow.com/a/5314685

Reply:
J'ai vu cette réponse pendant mes recherches mais je suis pas convaincu, ça me semble encore moins propre que ce que j'ai fait / plus complexe et relou niveau maintenance

ça implique du code dans le controller + passer systématiquement ça en param à toutes les méthodes de tout les mails + pareil pour les previews, bref ça clutter le code

là on set en dur une fois dans le ActionMailer et c'est good, seul downside c'est pas super dynamique (et j'ai répété ça de manière pas super DRY entre particulier et entreprise)

Reply:
> ça implique du code dans le controller + passer systématiquement ça en param à toutes les méthodes de tout les mails + pareil pour les previews, bref ça clutter le code

Pas d'après cette doc: https://edgeguides.rubyonrails.org/action_controller_overview.html#default-url-options

L'avantage c'est que tu peux faire ce que tu as suggéré https://github.com/etalab/admin_api_entreprise/pull/1255#discussion_r1363347364

Reply:
https://api.rubyonrails.org/classes/ActionMailer/Base.html

Section `Generating URLs`

> You can also define a `default_url_options` method on individual mailers to override these default settings per-mailer.

Cela m'étonne du coup.

Après, y'a plus simple:

```ruby
default_url_options[:host] = Rails.env.production? 'https://entreprise.api.gouv.fr' : "https://#{Rails.env}.entreprise.api.gouv.fr"
```